lib/cli/config.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * Responsible for loading / finding Mocha's "rc" files.
+ * This doesn't have anything to do with `mocha.opts`.
+ */
+
+const fs = require('fs');
+const findUp = require('findup-sync');
+const path = require('path');
+const debug = require('debug')('mocha:cli:config');
+
+/**
+ * These are the valid config files, in order of precedence;
+ * e.g., if `.mocharc.js` is present, then `.mocharc.yaml` and the rest
+ * will be ignored.
+ * The user should still be able to explicitly specify a file.
+ */
+exports.CONFIG_FILES = [
+  '.mocharc.js',
+  '.mocharc.yaml',
+  '.mocharc.yml',
+  '.mocharc.json'
+];
+
+/**
+ * Parsers for various config filetypes.  Each accepts a filepath and
+ * returns an object (but could throw)
+ */
+const parsers = (exports.parsers = {
+  yaml: filepath =>
+    require('js-yaml').safeLoad(fs.readFileSync(filepath, 'utf8')),
+  js: filepath => require(filepath),
+  json: filepath =>
+    JSON.parse(
+      require('strip-json-comments')(fs.readFileSync(filepath, 'utf8'))
+    )
+});
+
+/**
+ * Loads and parses, based on file extension, a config file.
+ * "JSON" files may have comments.
+ * @param {string} filepath - Config file path to load
+ * @returns {Object} Parsed config object
+ * @private
+ */
+exports.loadConfig = filepath => {
+  const ext = path.extname(filepath);
+  let config = {};
+  try {
+    if (/\.ya?ml/.test(ext)) {
+      config = parsers.yaml(filepath);
+    } else if (ext === '.json') {
+      config = parsers.json(filepath);
+    } else if (ext === '.js') {
+      config = parsers.js(filepath);
+    }
```

Comment:
Not sure if `yargs` already provides but, could be handy from `debug` PoV to have means to trace what pathnames _were_ tried but didn't exist.

Reply:
good idea.

- [ ] add debugs here

Reply:
this is actually handled by `findup-sync`.  there isn't any place we could hook in to.